src/source/Signaling/StateMachine.c

```diff
@@ -273,6 +272,11 @@ STATUS executeGetTokenSignalingState(UINT64 customData, UINT64 time)
                                                                             SIGNALING_CLIENT_STATE_GET_CREDENTIALS));
     }
 
+    // reset credentials expiration if we already have one.
+    if (NULL == pSignalingClient->pAwsCredentials) {
+        pSignalingClient->pAwsCredentials = 0;
```

Comment:
If it is NULL, we are setting it to 0? What is the purpose of this?

Reply:
Whoops, it's supposed to be pSignalingClient->pAwsCredentials->expiration = 0;